### PR TITLE
Upgrade macOS build to the new macos-14 runner

### PR DESCRIPTION
**Pull request by nobugshere**
This PR makes the following changes to the macOS yaml file:
- Use the new macos-14 runner (required adding sudo to make install)
- Xcode 15.3

(Other changes have been split into their own PRs.)

### CODE REVIEW

File: src/ui/wxWidgets/YubiCfgDlg.cpp
Comment (ronys) [inline src/ui/wxWidgets/YubiCfgDlg.cpp:1]: This change will need to be tested on the other wx platforms. Hope we don't have to #ifdef around specific wx versions to make this work.

Comment (nobugshere) [reply src/ui/wxWidgets/YubiCfgDlg.cpp:1]: A fair point.  In reality, the workaround was only ever needed on Mac, in retrospect it should have been an #ifdef all along.  I just did a quick check on Fedora 39 (switching between wx 3.0.5 and 2.3.4 via alternatives) and on an M1 Mac under Rosetta, both sized the text box correctly.

Comment (nobugshere) [reply src/ui/wxWidgets/YubiCfgDlg.cpp:1]: Oh.  I just realized you're using 3.2.2.1 in Flatpak.  I don't have a set-up to build and test that.  I only tested with the WX versions pre-built for Fedora.

**Comment by ronys** [comment]
Hi,

This is still marked as a draft. Any progress?

**Comment by nobugshere** [comment]
I'm just waiting to see if they remove the "Beta" designation from the macos-14 runner. Maybe they aren't going to do that until they move it to macos-latest, next quarter.  Looks like they are still making changes to the included tools.
[https://github.com/actions/runner-images/issues/9255](https://github.com/actions/runner-images/issues/9255)
[https://github.com/actions/runner-images/pull/9380](https://github.com/actions/runner-images/pull/9380)

In the mean time, I was thinking, at some point, I should rebase this branch to test the build with all the UI changes.  What are your thoughts?

**Comment by ronys** [comment]
No rush, just wanted to make sure this wasn't "orphaned".
Rebase is probably a good idea.

**Comment by nobugshere** [comment]
Rebased onto master commit 50815ab

**Comment by nobugshere** [comment]
Rebase and separated some commits into their own PRs.

**Review by Karimsultan — APPROVED** [review]

**Review by Karimsultan — APPROVED** [review]

**Comment by nobugshere** [comment]
The macos-14 runner image is GA as of today.  I think this is ready to merge.
https://github.blog/changelog/2024-04-01-macos-14-sonoma-is-generally-available-and-the-latest-macos-runner-image/

**Comment by ronys** [comment]
Thanks!

**Comment by ronys** [comment]
Hopefully late April / early May

**Comment by seamusdemora** [comment]
@ronys : Thanks for the information.  Could I ask why this process takes so long?

**Comment by ronys** [comment]
@seamusdemora - the process doesn't take long at all. My free time is the bottleneck here.
If you wish, you can grab a dmg generated as an artifact from the latest mac-pwsafe workflow run (https://github.com/pwsafe/pwsafe/actions/workflows/macos-latest.yml)
For example, https://github.com/pwsafe/pwsafe/actions/runs/8581201155 - click on the download icon in the lower right.

**Comment by seamusdemora** [comment]
Thanks! I'll try this.

**Pull request merged**